code/modules/materials/Mat_MaterialProcs.dm

```diff
@@ -396,6 +396,10 @@ triggerOnEntered(var/atom/owner, var/atom/entering)
 		payload.volume = R_IDEAL_GAS_EQUATION * T20C / 1000
 
 		if(agent_b && air.temperature > 500 && air.toxins > MINIMUM_REACT_QUANTITY )
+			// excess nitrogen prevents oxygen B from being created
+			if (air.nitrogen > air.toxins)
+				return
+
```

Comment:
This makes it even more difficult to generate Oxygen Agent B, it now specifically needs to be in an unusual atmosphere.